Adapter/AlgoliaElasticSearchAdapter.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Algolia\AlgoliaSearchElastic\Adapter;
+
+use Algolia\AlgoliaSearch\Helper\AdapterHelper;
+use Magento\Elasticsearch\SearchAdapter\Adapter as ElasticSearchAdapter;
+use Magento\Elasticsearch\SearchAdapter\Aggregation\Builder as AggregationBuilder;
+use Magento\Elasticsearch\SearchAdapter\ConnectionManager;
+use Magento\Elasticsearch\SearchAdapter\Mapper;
+use Magento\Elasticsearch\SearchAdapter\ResponseFactory;
+use Magento\Elasticsearch\SearchAdapter\QueryContainerFactory;
+use Magento\Framework\Search\RequestInterface;
+
+class AlgoliaElasticSearchAdapter extends ElasticSearchAdapter
+{
+
+    /** @var AdapterHelper */
+    private $adapterHelper;
+
+    /** @var QueryContainerFactory */
+    private $queryContainerFactory;
+
+    /**
+     * AlgoliaElasticSearch5Adapter constructor.
+     * @param ConnectionManager $connectionManager
+     * @param Mapper $mapper
+     * @param ResponseFactory $responseFactory
+     * @param AggregationBuilder $aggregationBuilder
+     * @param QueryContainerFactory $queryContainerFactory
+     * @param AdapterHelper $adapterHelper
+     */
+    public function __construct(
+        ConnectionManager $connectionManager,
+        Mapper $mapper,
+        ResponseFactory $responseFactory,
+        AggregationBuilder $aggregationBuilder,
+        QueryContainerFactory $queryContainerFactory,
+        AdapterHelper $adapterHelper
+    ) {
+
+        parent::__construct($connectionManager, $mapper, $responseFactory, $aggregationBuilder, $queryContainerFactory);
+
+        $this->adapterHelper = $adapterHelper;
```

Comment:
Don't you have to add "$this->queryContainerFactory = $queryContainerFactory;" here as well ?

Reply:
Sadly, the $queryContainerFactory variable in the parent class is a private one. In order to make that accessible, I had to this class to utilize it as there is not getter for this value :(